android/src/main/java/com/rudderstack/android/internal/infrastructure/LifecycleObserverPlugin.kt

```diff
@@ -22,15 +22,23 @@ import com.rudderstack.core.Analytics
 import com.rudderstack.core.ConfigDownloadService
 import com.rudderstack.core.Configuration
 import com.rudderstack.core.InfrastructurePlugin
+import com.rudderstack.core.Plugin
+import com.rudderstack.models.Message
 import java.util.concurrent.atomic.AtomicBoolean
 import java.util.concurrent.atomic.AtomicLong
 
 const val EVENT_NAME_APPLICATION_OPENED = "Application Opened"
 const val EVENT_NAME_APPLICATION_STOPPED = "Application Backgrounded"
 private const val MAX_CONFIG_DOWNLOAD_INTERVAL = 90 * 60 * 1000 // 90 MINUTES
 
+/**
+ * We use a [Plugin] instead of [InfrastructurePlugin] as sending events from Infrastructure plugin might
+ * not ensure all plugins to be ready
+ *
+ * @property currentMillisGenerator
+ */
 internal class LifecycleObserverPlugin(val currentMillisGenerator: (() -> Long) = { SystemClock.uptimeMillis() }) :
-    InfrastructurePlugin, LifecycleListenerPlugin {
+    Plugin, LifecycleListenerPlugin {
```

Comment:
This Interface change has broken the `LifecycleObserverPluginTest` test cases and we are getting following errors:
> Unresolved reference: shutdown.

Please do the required refactoring!

Reply:
Not getting this error. Can you recheck? If this exists still, we can get on a call to resolve.

Reply:
This issue doesn't seem to occur now.